STAC-24630: Add stackgraph backup v2 documentation [DRAFT] (#360)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/data-management/backup_restore/backup_restore.adoc

If a restore running without `--background` is interrupted (for example, by Ctrl+C), you must run `check-and-finalize` to scale up deployments and clean up resources.
====

== Configuration and topology data v2 (StackGraph)

> SUSE Observability {next-release-version} introduces v2 of StackGraph backup/restore. Both v1 and v2 backups are created until v2 is considered stable, after which v1 backups will be discontinued.

Reviewer: ```suggestion
{stackstate-product-name} {next-release-version} introduces v2 of StackGraph backup/restore. Both v1 and v2 backups are created until v2 is considered stable, after which v1 backups will be discontinued.
```
Are there any parameters that we might want to suggest keeping an eye on to ensure the stability of v2? Or is this something the customers will always know?

> Since SUSE Observability {next-release-version} we are rolling out v2 of StackGraph backup/restore. For now both v1 and v2 backups will be made, until v2 is deemed completely stable, at which point we'll sunset v1 backups.

Reviewer 2: `For now both v1 and v2 backups will be made` I read it as both types of backups are enabled, whereas in fact they both supported but only one of the can be enabled at a time. Might it be misleading for the others?

Author: Actually my plan here is to, when releasing, to enable taking both v1 and v2 backups.

> Since SUSE Observability {next-release-version} we are rolling out v2 of StackGraph backup/restore. For now both v1 and v2 backups will be made, until v2 is deemed completely stable, at which point we'll sunset v1 backups.

Reviewer 2: First-person voice + jargon: "we are rolling out" and "we'll sunset v1 backups". Suggest: "SUSE Observability {next-release-version} introduces v2 of StackGraph backup/restore. Both v1 and v2 backups are created until v2 is considered stable, after which v1 backups will be discontinued."

Author: applied

> V2 backups improve backup and restore speed and add incremental backups. Also, the StackGraph v2 restore procedure allows for restoring historical (backfill) data while the system is running, which means much quicker time to get the system back up.

Reviewer: ```suggestion
V2 backups improve backup and restore speed and add incremental backups. Also, the StackGraph v2 restore procedure allows restoring historical (backfill) data while the system is running resulting in quicker system backups.
```

> V2 backups offer significant improvements in backup and restore speed, as well as incremental backups. Also, the StackGraph v2 restore procedure allows for restoring historical (backfill) data while the system is running, which means much quicker time to get the system back up.

Reviewer 2: Tone down the marketing phrasing "offer significant improvements". Suggest: "v2 backups improve backup and restore speed and add incremental backups." Also note "V2" here vs "v2" elsewhere — standardize on lowercase `v2`.

Author: fixed

=== List StackGraph v2 backups

> To list the StackGraph V2 backups, execute the following command:

Reviewer 2: ```suggestion
To list the StackGraph v2 backups, execute the following command:
```

Author: fixed

> To list the StackGraph V2 backups, execute the following command:

Reviewer 2: Capitalization: "StackGraph V2" should be "StackGraph v2" to match the rest of the document.

Author: fixed

[,bash]
----

> sts-backup stackgraph-v2 list --namespace <NAMESPACE>

Reviewer 3: is the namespace argument optional?  We can default to `suse-observability`

----

Replace `<NAMESPACE>` with the namespace where {stackstate-product-name} is installed.

The output should look like this:

[,bash]
----

> Setting up port-forward to suse-observability-minio:9000 in namespace <NAMESPACE>...

Reviewer 3: is the `minio` still there?  I thought we switched to `s3proxy`?

✓ Port-forward established successfully
Listing Stackgraph backups in bucket 'sts-stackgraph-backup'...
NAME                               LAST MODIFIED
sts-backup-20251128-0300.graph.v2  2025-11-28 03:08:42 UTC
----

The timestamp when the backup is taken is part of the backup name.

=== Restore a StackGraph v2 backup

> The restore has a 'live' and 'backfill' portion of the backup. The 'live' part takes down the system and will restore the latest data. After the 'live' part is done, the system will scale up and is accessible by end-users. The `backfill` portion will be restored while the system is running. The following message appears to signify when the system is accessible again:

Reviewer 2: Two nits:
- "takes down the system" is informal — prefer "takes the system offline".
- Quoting is inconsistent: `'live'` / `'backfill'` are single-quoted here but `` `backfill` `` is in backticks in the same paragraph. Use backticks for literals throughout to match the rest of the doc.

Author: fixed

[,bash]
----
The system is back up and accessible. Continuing with backfilling historical data, this will happen while the system is running.
During this process not all historical data might be available.
----

To restore a StackGraph v2 backup, use one of the following commands:

==== Restore a specific backup

[,bash]
----
sts-backup stackgraph-v2 restore --namespace <NAMESPACE> --archive <BACKUP_NAME>
----

==== Restore the latest backup

[,bash]
----
sts-backup stackgraph-v2 restore --namespace <NAMESPACE> --latest
----

==== Additional flags

> * `--yes` or `-y` - Skip confirmation prompt (useful for automation)

Reviewer 2: The CLI (backup-cli#32, `cmd/stackgraphv2/restore.go`) defines a `--skip-stackpacks` flag ("Skip restoring stackpacks backup") but it isn't documented here. Consider adding it to this list.

Author: it is not documented with the other comments, it is also not really useful to the end-user so i'll leave it out

[NOTE]
====
If a restore running is interrupted (for example, by Ctrl+C), you must run `check-and-finalize` to scale up deployments and clean up resources. If the backfill phase of restoring was not started yet, also `sts-backup stackgraph-v2 backfill` must be run to complete restoring `backfill` data.
====

==== Additional commands

> * `sts-backup stackgraph-v2 backfill`. The restore command will automatically restore both the `live` and `backfill` portion of a backup. If, however, the restore command gets interrupted (through CTRL-C) or fails due to instability the cluster, this command can be used to restart the backfill job, restoring the `backfill` portion of the data.

Reviewer 2: Grammar + consistency:
- "fails due to instability the cluster" → "fails due to instability of the cluster".
- "CTRL-C" — the existing doc uses "Ctrl+C"; standardize on that.

Author: fixed

> * `sts-backup stackgraph-v2 abort`. If restoring the `backfill` portion of a backup cannot be recovered or is lost somehow, `abort` can be used to wrap up the restore command with the data that got already ingested successfully. This command is really a last resort and should not be needed in normal operation.

Reviewer 2: Informal phrasing: "This command is really a last resort" → "This command is a last resort".

Author: fixed

> * `sts-backup stackgraph-v2 check-and-finalize`. When any command (`restore`/`backfill`/`abort`) gets interrupted with Ctrl-C, the `check-and-finalize` command can be used to keep track of its progress, incling a `--wait` option which allows waiting for the job to finish.

Reviewer 2: Typo + consistency:
- "incling a `--wait` option" → "including".
- "Ctrl-C" → "Ctrl+C" to match the existing doc.

Author: fixed
